pkg/csi/service/wcpguest/controller.go

```diff
@@ -1371,27 +1371,32 @@ func (c *controller) GetCapacity(ctx context.Context, req *csi.GetCapacityReques
 	log := logger.GetLogger(ctx)
 	log.Infof("GetCapacity: called with args %+v", *req)
 
+	// Setting capacity to MaxInt64 for all topologies except for those which have been marked for deletion by VI Admin.
+	totalcapacity := int64(math.MaxInt64)
+	maxvolumesize := int64(math.MaxInt64)
+
 	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.WorkloadDomainIsolationFSS) {
-		var totalcapacity, maxvolumesize int64
 		zonesMap := commonco.ContainerOrchestratorUtility.GetZonesForNamespace(c.supervisorNamespace)
 		if req.AccessibleTopology != nil {
+			if zonesMap == nil {
+				return nil, logger.LogNewErrorCode(log, codes.Internal, "the guest cluster seems to be topology "+
+					"aware but CSI controller could not find zone instances in supervisor cluster.")
+			}
+
 			segments := req.AccessibleTopology.GetSegments()
 			if _, exists := zonesMap[segments["topology.kubernetes.io/zone"]]; !exists {
 				totalcapacity = 0
```

Comment:
Can we print a log when zone is not found in the map and mention in the log that capacity in the response will be 0

Reply:
Added Info log for `Zone %q is either marked for deletion or is not part of the namespace %q. Setting capacity to 0.`